incubator/nginx/nginx.libsonnet

```diff
@@ -0,0 +1,149 @@
+local k = import 'ksonnet.beta.2/k.libsonnet';
+local deployment = k.extensions.v1beta1.deployment;
+local container = deployment.mixin.spec.template.spec.containersType;
+
+{
+  parts:: {
+    deployment:: {
+      local defaults = {
+        imageTag: "1.10.2-r3",
+        imagePullPolicy: "IfNotPresent",
+      },
+
+      noVhost(namespace, name, labels={app: name})::
+        base(namespace, name, labels),
+
+      hasVhost(namespace, name, labels={app: name})::
+        local volume = {
+          name: "nginx-vhost",
+          configMap: { name: name },
+        };
+        local dataMount = {
+          name: "nginx-vhost",
```

Comment:
We probably want to pass this in as a parameter that defaults to `"nginx-vhost"`, since it refers to the configmap's name.